src/context.rs

```diff
@@ -0,0 +1,107 @@
+use crate::{InternalError, SignatureError};
+
+/// Ed25519 contexts as used by Ed25519ctx and Ed25519ph.
+///
+/// Contexts are domain separator strings that can be used to separate uses of
+/// the protocol between different protocols (which is very hard to reliably do
+/// otherwise) and between different uses within the same protocol.
```

Comment:
Is this what I provided earlier? It's adapted from the security considerations section of RFC8032, but looking at it now it seems rather redundant: "separator strings... separate... the protocol between different protocols"

Maybe this?

```suggestion
/// Contexts are domain separator strings that can be used to isolate uses of
/// the algorithm between different protocols (which is very hard to reliably do
/// otherwise) and between different uses within the same protocol.
```